Library/PTPusher.m

```diff
@@ -274,6 +274,20 @@ - (void)subscribeAll
   }
 }
 
+- (void)unsubscribeAllChannels
+{
+  if (channels && channels.count) {
+	NSMutableArray *channelsToUnsubscribe = [[NSMutableArray alloc] init];
+	  for (PTPusherChannel *channel in [channels allValues]) {
+		[channelsToUnsubscribe addObject:channel];
```

Comment:
I'm also don't see the purpose of checking 'channels' (it's created on initialisation) or its count. If the count is zero, then allValues will return an empty array anyway.